### PR TITLE
Lldb testing

### PR DESCRIPTION
**Pull request by adrian-prantl**
[build-script-impl: Have LLDB honor -llvm-lit-args](https://github.com/apple/swift/commit/09242f1969e7d7cfd59d5b9163ea565752166d91)

[Add an LLDB incremental buildbot preset](https://github.com/apple/swift/commit/1bfac91592d979406a9edd142a94d0cef99ae228)

### CODE REVIEW

**Comment by adrian-prantl** [comment]
@swift-ci test

File: utils/build-script-impl [hunk 1]
```diff
@@ -3554,7 +3554,7 @@ function build_and_test_installable_package() {
               TIMEOUT_ARGS=--timeout=600 # 10 minutes
             fi
             with_pushd "${PKG_TESTS_SOURCE_DIR}" \
-                call python3 "${LIT_EXECUTABLE_PATH}" . -sv --param package-path="${PKG_TESTS_SANDBOX}" --param test-exec-root="${PKG_TESTS_TEMPS}" --param llvm-bin-dir="${LLVM_BIN_DIR}" ${TIMEOUT_ARGS}
+                call python3 "${LIT_EXECTABLE_PATH}" . -sv --param package-path="${PKG_TESTS_SANDBOX}" --param test-exec-root="${PKG_TESTS_TEMPS}" --param llvm-bin-dir="${LLVM_BIN_DIR}" ${TIMEOUT_ARGS}
```
Comment (JDevlieghere) [inline utils/build-script-impl:3557]: Typo?

Comment (JDevlieghere) [reply utils/build-script-impl:3557]: Also should we still have the `-sv` in here?

Comment (adrian-prantl) [reply utils/build-script-impl:3557]: I shouldn't have touched this at all in this patch. That said, the proliferation of different lit invocations in this file is a mess.

**Comment by adrian-prantl** [comment]
@swift-ci test

**Comment by adrian-prantl** [comment]
@swift-ci test package

**Comment by adrian-prantl** [comment]
@shahmishal does that preset look reasonable to you?

**Review by JDevlieghere — APPROVED** [review]
Thanks for doing this

**Comment by edymtt** [comment]
Looks good to me as well.

Small doubt: is it expected that this PR targets `rebranch` instead of `main`?

**Comment by adrian-prantl** [comment]
> Looks good to me as well.
> 
> Small doubt: is it expected that this PR targets `rebranch` instead of `main`?

I also cherry-picked this to main. It's just that I spend most of my work on rebranch these days, that's why it showed up here first.

**Pull request merged**